src/Ex5.java

```diff
@@ -0,0 +1,67 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Ex5 {
+
+        public static void main(String[] args) {
+
+            String name = "Дарья Михальченко";
+            char[] stringToCharArray = name.toCharArray();
+            String tempo = "***** ***********";
+            String str = "";
+            List<String> used = new ArrayList<>();
+            List<Integer> idexies_of_matches = new ArrayList<Integer>();
+            Boolean bool = false;
+
+            System.out.println("Вам нужно угадать мое имя и фамилию: " + tempo);
+
+            Scanner in = new Scanner(System.in);
+
+            while(tempo.contains("*")){
+                System.out.println("Введите букву:");
+                char litera = in.nextLine().charAt(0);
+
+                if (Character.toString(litera).equals(" ")) {
+                    System.out.println("Неверно!");
+                }
+            /* if(Character.toString(litera).trim().length() == 0){
+                System.out.println("Вы не ввели ни одного символа!");
+            }*/
+
+                if(used.contains(Character.toString(litera))){
+                    System.out.println("Вы уже вводили эту букву");
+                } else {
+                    used.add(Character.toString(litera));
+                    char[] myNameChars = tempo.toCharArray();
```

Comment:
you can store hidden name as chars initially as array or List. You will save memory and time by not converting every time your string to char array and back